internal/block/block_pool_test.go

```diff
@@ -244,6 +294,37 @@ func (t *BlockPoolTest) TestBlockPoolCreationFailsWhenGlobalMaxBlocksIsZero() {
 	assert.ErrorContains(t.T(), err, CantAllocateAnyBlockError.Error())
 }
 
+func (t *BlockPoolTest) TestTryGetWhenLimitedByGlobalBlocks() {
+	bp, err := NewGenBlockPool(1024, 10, semaphore.NewWeighted(2), createBlock)
+	require.Nil(t.T(), err)
+	// 2 blocks can be created.
+	b1, err1 := bp.TryGet()
+	require.Nil(t.T(), err1)
+	require.NotNil(t.T(), b1)
+	b2, err2 := bp.TryGet()
+	require.Nil(t.T(), err2)
+	require.NotNil(t.T(), b2)
+
+	b3, err3 := bp.TryGet()
+
+	require.Nil(t.T(), b3)
+	require.NotNil(t.T(), err3)
+	require.ErrorIs(t.T(), err3, CantAllocateAnyBlockError)
+	require.Equal(t.T(), int64(2), bp.totalBlocks)
+}
+
+func (t *BlockPoolTest) TestTryGetWhenTotalBlocksEqualToMaxBlocks() {
+	bp, err := NewGenBlockPool(1024, 10, semaphore.NewWeighted(2), createBlock)
```

Comment:
```suggestion
	bp, err := NewGenBlockPool(1024, 10, semaphore.NewWeighted(20), createBlock)
```
it will be limited by global blocks otherwise